app/hid/keycodes.py

```diff
@@ -14,6 +14,8 @@
 MODIFIER_RIGHT_SHIFT = 1 << 5
 MODIFIER_RIGHT_ALT = 1 << 6
 MODIFIER_RIGHT_META = 1 << 7
+# German AltGr Key right of spacebar
+MODIFIER_ALT_GR = MODIFIER_LEFT_CTRL | MODIFIER_LEFT_ALT
```

Comment:
The extra definition is necessaray since `Alt Gr` isn't a real key following the usb hid standard. I explained the interpretation TinyPilot does when pressing `Alt Gr` on a German Layout in my original PR (look above). It is clearly a difference in Hex Value transmitted to `Righ Alt`. German Layout keyboards just "emulate" the press of `Left Ctrl` + `Left Alt` when pressing `Alt Gr`. The confusion happens when the `Alt Gr` key is at the same position where the `Right Alt` Key on US-Keyboards is. German Keyboards simply don't have any `Right Alt` Key. Did you use a "real" German Layout Keyboard for the tests? I guess it could be that pressing `Right Alt` on a US-Layout then translates to `Alt Gr` on the guest system with Windows set to German Layout, hence the correct output. 
Even before my commits the normal TinyPilot web UI worked with pretty much all the special characters for german keyboards, when the host and guest system have Windows set to German Layout. Pasting was the only problem for german layouts.
Another Question: 
Did the `\` come through when pasting? On german layout it is produced via `Alt Gr` + `ß` like the `{` is produced via `Alt Gr` + `7`